### ISSUE TITLE
 Think twice before giving feedback to this community!

### ISSUE THREAD

**Issue by JuanDidNothingWrong4** [BastiaanOlij/godot_openvr#21]
(Since Noshyaar keeps closing this, I am going to continue to post it until one thread remains visible for discussion, no need to censor)

![](https://i.gyazo.com/364f1b11bc57924374d0a8c81df1eeab.png)

Lead developer sent this message after a user was giving his honest feedback via this [reddit thread](https://www.reddit.com/r/gamedev/comments/7gmtxl/godot_30_beta_1_is_there_right_in_time_for_ludum/dqk9nel/).

They also said (I kid you not) -- giving feedback is "**bullying**",  a "**professional whiner**", and "**ranting**". Reduz's passive aggressive tone and insulting behavior is rampant there.  I already terminated my patreon account because of this, and not saying you should either (FLOSS projects) are great, however, for the lead developer to act like this towards people giving feedback.. is not okay!

Let me re-iterate, not saying that you should stop giving feedback, but I just want this message to be seen across github and reach as far as possible. Because this kind of communication and how the lead developer treats a regular Godot developer is not right at all. I've never seen this behavior from a FLOSS project lead developer.  It's a shame, because the guy giving feedback is just a random person anyway, and the way the project managers and lead developers took it so personally like it was an attack (when it wasn't), means the lead developers/contributors are possibly not viewing your issues with the right mindset.  

Also, after reading the last paragraph from [this post by akien](https://www.reddit.com/r/gamedev/comments/7gmtxl/godot_30_beta_1_is_there_right_in_time_for_ludum/dqlqj2v/)
> From here on, we will avoid responding to xxx's threads on the various Godot communities, and focus instead on the huge amount of constructive feedback and bug reports that we get from other users. This will likely overlap with some of the issues mentioned above, which will thus be fixed eventually, but we won't accept being bullied into working on them now when we have different priorities. Users are free to disagree with our priorities, but in the end the contributors are the ones making the call.

He's saying they will now favor/prioritize issues based on the user reporting it, and not the issue itself. (Extreme prejudice), which will not be good for the project because the issue should be irrelevant to the user reporting it. Just because if a user who has a snotty attitude that is reporting a huge client crash error, doesn't mean that client crash error should not be taken lightly.

He's also stating that anyone giving hard feedback is now "bullying" them. Which is not true whatsoever.

Just wanting to post this to get the word out there, hopefully some things change. @reduz @akien-mga  This is just unbelievable :(

**Comment by JuanDidNothingWrong4** [comment]
@BastiaanOlij Sad you support a lead developer who treats people who give feedback like human garbage.

**Comment by BastiaanOlij** [comment]
I think you need to take a look in the mirror and reflect on the last few months. Reduz and Akien have been nothing but civil to everyone for as long as I've know them. They have kept their cool with even the most difficult characters that have popped up from time to time. Their reactions when you first started your complaints were civil and on point. That you have worn them down over the past few month for them to finally react with words that could possibly have been better chosen just proves they are human beings.

**Comment by BastiaanOlij** [comment]
Case in point is this exact repository. When I first appeared out of nowhere with a wish to do VR I approached the community, asked whether VR was being considered, as that is what I want to be working on.

The reaction at the point was simple, Juan had been given some hardware but was tied up in the new rendering engine. They wanted to look at VR at some undetermined point in the future. And that was it.

I could have reacted in two ways. I could have reacted like you, and told them how stupid they were because no self respecting engine should pass on VR, that is the future, that the fact that Juan had been given hardware means that he must drop everything and do VR.

Or I could have reacted in the way I chose to react. Accept that they have their hands full, that they are juggling many, MANY, priorities. That their time is limited and that there is plenty to do before they are ready and able to tackle an issue that is close to what I want to use Godot for.

Given, I'm in a position not many are. First off this being a hobby and second having the skills to actually implement the feature I saw so lacking. Still that does not mean that if I was not in that position, that would somehow give me any rights to hammer people for months in the way you have.

**Comment by JuanDidNothingWrong5** [comment]
@BastiaanOlij Read the reddit thread sir. Juan's response to that user giving his honest feedback was unprofessional.  

> Reduz and Akien have been nothing but civil

True to an extent, but not until he messaged that person calling him a moron. That's not civil at all.

Albeit, I appreciate u actually not closing this/deleting it like they are. I just want you to understand, Juan is not the end all perfect being. That kind of response from a lead developer should not be tolerated at all, by you or any contributor. 

And akien already apologized on his behalf,  it's just sad. There is just no reason to treat people who give feedback like that.

**Comment by BastiaanOlij** [comment]
There is a long, LOOOOOOOOOOONG history before that reddit thread.

I've known Juan and Remi for awhile now, they have been pushed to the brink and are reacting like humans do after such a long barrage. It is uncalled for, it is unjustified. That Juan and Remi may have _finally_ overreacted, I don't hold that against them at all. I would have cracked long before they did.

I don't know who the redditers are, I don't know who you are, but nothing justifies the crusade you guys are on.

**Comment by JuanDidNothingWrong5** [comment]
Doesn't matter what the history is at all, lead developers should not be calling reddit users a "moron" over giving their feedback. Continue with more red-herrings though.

**Comment by JuanDidNothingWrong5** [comment]
My crusade is about the lead developer treating the reddit user like human garbage after giving feedback. If that reddit user is a troll (his points about the engine, doesn't seem like he's trolling, seems like he's ranting), but that's just an opinion. That doesn't give the lead developer the green light to call him a "moron".

**Comment by JuanDidNothingWrong5** [comment]
> It is uncalled for, it is unjustified. That Juan and Remi may have finally overreacted, I don't hold that against them at all. I would have cracked long before they did.

So if they overrated and you admit it was wrong, you don't "hold that against them at all". So basically, it's justified? You are making absolutely zero sense.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 Of course my reply was unprofessional. We are not running a company here. We are individuals working on their own acting like individuals. 

I didn't call anyone moron though, nor treated this guy like human garbage. I think you are exaggerating or overly sensitive.

**Comment by JuanDidNothingWrong5** [comment]
@reduz

Wrong, you look at the screenshot. You implied he's a moron, and threatened to ban him over giving feedback. Continue lying, it'll get you no where.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 you say "the lead developer treating the reddit user like human garbage", I am the lead developer, and I did not treat this user like human garbage. I don't think anyone who was looked at that image thinks that the user has been treated like garbage either, save for you.

**Comment by BastiaanOlij** [comment]
And that justifies your actions?

The history is all important. This guy pushed and pushed and pushed and pushed. Right or wrong, at some point people have had enough. That was probably exactly the reaction he was waiting for. Bait a trap and snap.

If you take the time to get to know Juan, you'll see he is one of the most helpful and respectful development leads I've come across and I've been around for awhile. I've seen him take hours out of his schedule to help people that in other projects would have been ignored or dismissed off hand. 

He has been nothing but reasonable from the start but the guy just won't see reason.
To have him have his fill, you have to be a special kind of person to have pushed him that far.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 That user posted on Reddit literally "So many issues still wrong with Godot, it's kind of pathetic most of it actually gets noticed by the developers (but not taken seriously)."

To you that is fine? this user can call what we do pathetic and we have to sit and accept it? I was simply sending a message to that user, in PRIVATE (because my intention was, in fact, NOT shaming him, as community hates him already) and telling him to please stop that attitude, because insults are not tolerated. This is very far from treating somebody as garbage. In fact, I repeated many times that I don't mind he still sticks around.

**Comment by JuanDidNothingWrong5** [comment]
> If you take the time to get to know Juan, you'll see he is one of the most helpful and respectful development leads I've come across and I've been around for awhile. I've seen him take hours out of his schedule to help people that in other projects would have been ignored or dismissed off hand.

This is true. And from what I have seen he is helpful and generally nice to the community.  However, that does not excuse or give him the green light to call a user a moron over facebook.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 In fact, that user (MakeGodotGreatAgain) still keeps talking nice about Godot in reddit if you check his history. I think he understood the message fine and the I don't think he is a moron, it's you who for some reason does not understand it.

**Comment by JuanDidNothingWrong5** [comment]
> This is very far from treating somebody as garbage

> Calls someone a moron in a facebook message after they gave their feedback about the engine

Pick one.

@reduz That's all you read from his post? Did you not read the last paragraphs where he talked about you in a good light and said you're helpful?  To me, it seemed like he was ranting, but giving honest feedback. And your message to him didn't seem appropriate at all.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 I know he was ranting, but that's not the right way to rant. Even if he is more respectful towards the end. You can't preach for others to be respectful of you if you don't respect others first.

**Comment by BastiaanOlij** [comment]
> This is true. And from what I have seen he is helpful and generally nice to the community. However, that does not excuse or give him the green light to call a user a moron over facebook.

Neither does it give you the right to have a defamation campaign against him. Your actions so far have been far worse.

When things get heated, sometimes people over react.

> To me, it seemed like he was ranting, but giving honest feedback

That would be true the first time he tried to get the message out. But after not wanting to understand the very valid reactions given and bringing the topic back up over and over and over again, often with very negative innuendos, at some point enough is enough.

**Comment by JuanDidNothingWrong5** [comment]
> You can't preach for others to be respectful of you if you don't respect others first.

 Says the one who's called him a moron in a facebook message. You were unprofessional in what you did, I know it's hard to admit it because your ego is so high, but it's becoming pretty sad.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 I still don't understand why you are doing this, this user does not really seem to care, I am not angry with that user and i don't think he is garbage, and the entire community does not seem to agree with what you are doing. If my Ego was high as you mention, I would be working at Epic or Unity.

**Comment by JuanDidNothingWrong5** [comment]
> Neither does it give you the right to have a defamation campaign against him. Your actions so far have been far worse.

Defamation? How about the user being called a moron by the lead developer, that's not defamation? As I said, you are not making any sense.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 you are the sole person in this universe since the big bang who adheres to me having called that person a moron. I don't think that person even thinks that.

**Comment by JuanDidNothingWrong5** [comment]
@reduz That person quite literally edited his post after he got your message and said "Am done with the Godot community" and he deleted his facebook (AFAIK).  Pretty sure you calling him a moron did not help.

I'm just amazed how you don't think what you said to him was wrong in anyway.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 In fact, all you achieved with this campaign is this person being called a moron by pretty much the whole community, because they think you are him. Big favor you did to him, much worse than anything you are claiming I did.

If someone destroyed the poor guy's reputation, that's you doing this madness.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 At least I sent him the message in private, because I did not want to harm his reputation, you literally blasted him off the planet.

**Comment by BastiaanOlij** [comment]
> Defamation? How about the user being called a moron by the lead developer, that's not defamation? As I said, you are not making any sense.

Uhm, that message was sent in private so no, that is not defamation. Uncalled for yes, not Juans finest moment sure, understandable? Yeah seeing the history, while it may not justify it, understandable certainly.

But definitely not an excuse to go on the tirade you've gone on.

I know it's a bit hard in this online world but in reality this would be a reason to buy each other a cup of coffee, have an honest discussion, make up, shake hands, and come out of it in an understanding of why things went the way they did.

**Comment by JuanDidNothingWrong5** [comment]
> Uhm, that message was sent in private so no, that is not defamation

Wrong. Just because a message was sent in private "over facebook" doesn't mean that person is immune to defamation. That's the most silliest thing I've ever read.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 Also, I do agree I should have used kinder words with him, but it was not a good time for him to rant as I was not in a good mood (I was getting surgery a couple hours later, i just got back home from the hospital). 

In fact, had him complained of having been offended I would have apologized. But now it's too late, I think you ruined him with this crusade.

**Comment by JuanDidNothingWrong5** [comment]
> Also, I do agree I should have used kinder words with him

There it is. That's all I wanted to hear. Took a while, though.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 If anything, I hope this helps resolve these kind of issues better in the future.

**Comment by BastiaanOlij** [comment]
> Wrong. Just because a message was sent in private "over facebook" doesn't mean that person is immune to defamation. That's the most silliest thing I've ever read.

Defamation by the litteral meaning of the word is ruining someones reputation by saying bad, possibly untruthful, things about a person in _public_

Which is what you are doing, which is not what Juan did.

Juan reacted in private to a long history of being pushed. Not in the right way, he could have handled that better, but as I've repeated many times, in a way that is understandable and all too human. I can't blame him for that. I've been there many times myself, I would have had my fill long before he did.

It is reason to grab a coffee together and talk out your differences. It is not a reason to post hundreds of messages all over the internet calling him out.

**Comment by BastiaanOlij** [comment]
Ah looks like we're starting to pull out those virtual coffees :)

**Comment by reduz** [comment]
@JuanDidNothingWrong5 Honestly I'm more worried about Noshyaar who claimed he was going to commit suicide and disappeard from the internet (and gthub), thanks to how nervous you got him. He was unrelated to this, and I hope he did nothing crazy, or you'll have to carry the weight.

**Comment by JuanDidNothingWrong5** [comment]
>and I hope he did nothing crazy, or you'll have to carry the weight.

Says the one who started it all by calling someone a moron after they gave feedback about **your engine**

**Comment by BastiaanOlij** [comment]
Ditto here.. waking up to that news this morning.. hope he's ok

**Comment by reduz** [comment]
@JuanDidNothingWrong5 the only moron here is you, I don't have any problem with MakeGodotGreatAgain.

**Comment by JuanDidNothingWrong5** [comment]
> the only moron here is you, I don't have any problem with MakeGodotGreatAgain.

Then why did you call him a moron on the facebook message and threaten to ban him. You obviously have a problem with him.

**Comment by JuanDidNothingWrong5** [comment]
Also, all my replies in this thread will be deleted after GitHub flags my account, so might want to archive.is if we ever gonna look back at this.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 I already explained, i admit I could have been kinder with him in words, but his tone for ranting was disrespectful. I don't mind him ranting, and never had a problem with him before (he's done it plenty of times in case you don't know), but last time was disrespectful.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 But i don't think he is a moron, you definitely are though.

**Comment by BastiaanOlij** [comment]
ok... that must have been a bad coffee :)

**Comment by reduz** [comment]
@JuanDidNothingWrong5 Keep all the logs you want, honestly only you care, not even him.

**Comment by JuanDidNothingWrong5** [comment]
Just got an archive.is link right now, and will report the archive thread to the software conservancy aswell.

Absolutely abhorrent behavior from you, Juan. Not only victim blaming me (the messenger), for the actions of one of your contributors, but also now calling me a moron. All after calling you out for your behavior (which you even said you could have used nicer words).

Really sickening behavior to be honest. Glad it's going to be logged.

**Comment by reduz** [comment]
@JuanDidNothingWrong5 You are the only one who thinks this in the whole universe, but go ahead and test if someone else cares.

**Comment by BastiaanOlij** [comment]
There are hundreds of posts from you plastered all over Github, deleted or not, that show you are far more then "just the messenger". I think people will understand what is going on here, and what is being said. While I don't agree with all choices of words here people understand provocation and people understand human nature. Anyone who bothers to read this from start to finish, and adds in the discussion on reddit, will quickly understand the context for the choice of words.

**Comment by JuanDidNothingWrong5** [comment]
@BastiaanOlij I'm not the one calling people morons.
